azext_iot/common/base_discovery.py

```diff
@@ -12,6 +12,8 @@
 from typing import Any, Dict, List
 from types import SimpleNamespace
 
```

Comment:
No changes to user facing help content to explain the added flexibility?

Reply:
I will change the param help for hub name and dps name to reflect that it can be hostname too. In auth type, I will include that if login and hostname are presented, look up will be skipped unless needed.

Reply:
![image](https://github.com/Azure/azure-iot-cli-extension/assets/73560279/34efd78b-fdbc-48c6-8378-29eafd9136c2)